hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/FileUtil.java

```diff
@@ -667,6 +670,9 @@ public static void unZip(InputStream inputStream, File toDir)
           if (!file.setLastModified(entry.getTime())) {
             numOfFailedLastModifiedSet++;
           }
+          if (entry.getPlatform() == ZipArchiveEntry.PLATFORM_UNIX) {
+            Files.setPosixFilePermissions(file.toPath(), permissionsFromMode(entry.getUnixMode()));
```

Comment:
what if this fails?

Reply:
If you fail, then leave the default permissions, also is not executable permissions, but for a Linux system, decompression process is going to this logic, if it is a Windows platform, you will skip this logic, because the window system is not using the Unix permissions, this is also consistent with the original way of relaxation.

Reply:
ok